File: python/programmers/pg_lv2_큰수만들기.py
def solution(number, k):
    stack = []
    deleteCount = 0
    
    for idx, i in enumerate(number):
        while len(stack) != 0 and stack[-1] < int(i) and deleteCount < k:
            stack.pop()
            deleteCount += 1
            
        stack.append(int(i))
        
        if deleteCount == k: 
            return "".join(map(str, stack)) + number[idx + 1:] if len(stack) != 0 else number[idx + 1:]
        
    remainK = k - deleteCount
    return number[:-remainK]

File: programmers/pg_lv2_큰수만들기.js
function solution(number, k) {
  const n = number.length;
  const target = n - k;
  let result = Array.from({ length: target }, (v, idx) => +number[idx + k]);

  for (let i = k - 1; i >= 0; i--) {
    let curNum = +number[i];
    for (let j = 0; j < target; j++) {
      if (curNum >= result[j]) {
        let tmpNum = result[j];
        result[j] = curNum;
        curNum = tmpNum;
      } else break;
    }
  }

  return result.join("");
}
